rawx/handler_chunk.go

```diff
@@ -275,24 +287,34 @@ func (rr *rawxRequest) checkChunk() {
 		if expected_hash == "" {
 			expected_hash = rr.chunk.ChunkHash
 		}
-		checksum, err := in.recomputeHash()
+		expected_hash = strings.ToUpper(expected_hash)
+
+		var filter io.ReadCloser
+		var in *io.LimitedReader
+		in, filter, err = rr.getChunkReader(chunkIn, rr.chunk.size, rangeInfo{})
 		if err != nil {
-			/* how check return code ? */
-			LogError("Fail to compute md5sum: %s", err)
-			rr.replyError(err)
+			rr.replyCode(http.StatusPreconditionFailed)
 			return
 		}
-		if !strings.EqualFold(checksum, expected_hash) {
-			LogError("Md5sum doesn't match: computed:%s, expected:%s",
-				checksum, expected_hash)
-			rr.replyCode(http.StatusPreconditionFailed)
+		defer filter.Close()
```

Comment:
When there is no compression, the filter is `nil`.
```suggestion
		if filter != nil {
			defer filter.Close()
		}
```